c10/core/TensorImpl.h

```diff
@@ -33,6 +33,11 @@ C10_DECLARE_bool(caffe2_keep_on_shrink);
 // respect caffe2_keep_on_shrink.
 C10_DECLARE_int64(caffe2_max_keep_on_shrink_memory);
 
+// Error message to show when the user tries to change tensor metadata on
+// Tensor created from .data or .detach().
+//
+// See NOTE [ Metadata Change for a Detached Tensor ] for details.
+extern C10_API const std::string err_msg_tensor_metadata_change_not_allowed;
```

Comment:
can this be in the class (static + protected)?  I can't see a case where this would be necessary outside the class, but maybe I'm missing it.